shared/domains/chelonia/chelonia.js

```diff
@@ -1684,12 +1684,14 @@ export default (sbp('sbp/selectors/register', {
       // All of these situations should trigger parsing the respinse and
       // conlict resolution
       if (response.ok || response.status === 409 || response.status === 412) {
-        const serializedData = await response.json()
-        currentValue = parseEncryptedOrUnencryptedMessage.call(this, {
-          contractID,
-          serializedData,
-          meta: key
-        })
+        const serializedDataText = await response.text()
+        currentValue = serializedDataText
+          ? parseEncryptedOrUnencryptedMessage.call(this, {
+            contractID,
+            serializedData: JSON.parse(serializedDataText),
+            meta: key
+          })
+          : undefined
```

Comment:
409 will return:

```js
return h.response(existing).etag(cid).header('x-cid', `"${cid}"`).code(409)
```

Which will still be truthy and therefore will be passed to `JSON.parse`.

Same for 412:

```js
      // We need this `x-cid` because HAPI modifies Etags
        return h.response(existing).etag(cid).header('x-cid', `"${cid}"`).code(412)
```

(boyscouting: that `// We` should be on the same line as `return`...)